src/mobile-token/utils.ts

```diff
@@ -18,5 +19,13 @@ export const isMobileToken = (remoteToken?: RemoteToken) =>
 export const getDeviceName = (remoteToken?: RemoteToken) =>
   remoteToken?.keyValues?.find((kv) => kv.key === 'deviceName')?.value;
 
+export const getDeviceNameForCurrentToken = (
+  activatedToken?: ActivatedToken,
+  remoteTokens?: RemoteToken[],
+) =>
+  getDeviceName(
+    remoteTokens?.find((token) => token.id === activatedToken?.getTokenId()),
```

Comment:
activatedToken doesn't have `keyValues` itself?

Reply:
No, it doesnt.